index.js

```diff
@@ -0,0 +1,117 @@
+fetch('CountriesData.json')
+    .then((response) => response.json())
+    .then((countries) => {
+        const countriesContainer = document.getElementById('countries-g');
+        if (!countriesContainer) {
+            console.error('countries-g container not found in the DOM!');
+            return;
+        }
+
+        // global veraible
+        let allCountries = countries;
+
+        // view the countries
+        function displayCountries(countriesToDisplay) {
+        countriesContainer.innerHTML = ''; // Cleaning all the current HTML in the div
+        countriesToDisplay.forEach((country) => {
+        const countryDiv = document.createElement('div');
+        countryDiv.className = 'country';
+            
+        const countryLink = document.createElement('a');
+        countryLink.href = `details.html?country=${country.name}`;
+        countryDiv.appendChild(countryLink);
+            
+        const countryflagDiv = document.createElement('div');
+        countryflagDiv.className = 'country-flag';
+            
+        const imgFlag = document.createElement('img');
+        imgFlag.src = `${country.flag}`;
+        imgFlag.alt = `${country.name}`;
+        countryflagDiv.appendChild(imgFlag);
+        countryLink.appendChild(countryflagDiv);
+
+        const countryInfo = document.createElement('div');
+        countryInfo.className = 'country-info';
+
+        const countryH = document.createElement('h2');
+        countryH.textContent = `${country.name}`;
+        countryInfo.appendChild(countryH);
+
+        const countryPopulation = document.createElement('p');
+        const strongP = document.createElement('strong');
+        strongP.textContent = 'Population: ';
+        countryPopulation.appendChild(strongP);
+        countryPopulation.appendChild(document.createTextNode(country.population));
+        countryInfo.appendChild(countryPopulation);
+
+        const countryRegion = document.createElement('p');
+        const strongR = document.createElement('strong');
+        strongR.textContent = 'Region: ';
+        countryRegion.appendChild(strongR);
+        countryRegion.appendChild(document.createTextNode(country.region));
+        countryInfo.appendChild(countryRegion);
+
```

Comment:
In this page you have a lot of similar code again an again, can you think about a way to avoid it?